munit-snap/src/main/scala/munit/SnapshotSuite.scala

```diff
@@ -0,0 +1,130 @@
+package munit
+
+import io.circe.*
+import io.circe.parser.*
+import io.circe.syntax.*
+import io.circe.syntax.given
+import io.{circe => ic}
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import scala.io.Source
+import scala.util.Try
+
+/**
+ * Provides snapshot testing capabilities for munit.
+ *
+ * Snapshot tests are great ways of preventing regressions in the outputs of complex processes.
+ * An example is compiler stage tree and context information. Another example is a complex web
+ * page rendering, or indentation engine output.
+ *
+ * Given a json encodeable output, the first time the test is run, the output is encoded and
+ * recorded to a file in the same directory as the test suite containing the snapshot test,
+ * using the filename of the test suite, the name of the test, and the line number of the test.
+ *
+ * The second time the test is run, the result is compared to the decoded result recorded in the
+ * file. Snapshot files should be stored with the test source code in version control
+ *
+ * To clear an outcome of a previous run, delete the corresponding json file from the folder.
+ */
+abstract class SnapshotSuite extends FunSuite:
+
+  given codec[A: Encoder: Decoder]: Codec[A] =
+    ic.Codec.from(summon, summon)
+
+  /**
+   * Given a json codec for the type of the result of the snapshot test body, execute a snapshot
+   * test on the body results.
+   *
+   * The first time the test is run, the json encoded result is recorded to a file in the same
+   * directory as the test file using the location, test name, and line number of the snapshot
```

Comment:
could we maybe have a directory `snapshot-files` or something where all the json files with the snapshotted values will exist? 
otherwise with the number of tests we have I think it will be a bit hard to navigate in the tests folder trying to find the `...Suite` ones.

Reply:
Sure. I'd also really like a Tag to allow for resetting a test. If you use .reset (like .only or .debug) the snapshot is deleted and the test passes and stores the new result.

Reply:
It might also be better to also store the hashes of the result. Compare them first before reading the file. Some files could get quite large.

Reply:
That tag sounds very useful. I'm looking forward to this work!

Reply:
>If you use .reset (like .only or .debug) the snapshot is deleted and the test passes and stores the new result.
that would be really nice yes. either in this PR or later I guess